src/Windowing/OpenToolkit.Windowing.Common/Input/KeyboardState.cs

```diff
@@ -314,27 +225,20 @@ private static void ValidateOffset(int offset)
             }
         }
 
-        /// <summary>
-        /// Compares two KeyboardState instances.
-        /// </summary>
-        /// <param name="other">The instance to compare two.</param>
-        /// <returns><c>true</c>, if both instances are equal; <c>false</c> otherwise.</returns>
-        public bool Equals(KeyboardState other)
+        private static (int intOffset, int bitOffset) GetOffsets(Key key)
         {
-            var equal = true;
-            unsafe
+            if (key <= Key.Unknown || key > Key.LastKey)
             {
-                var k2 = other._keys;
-                fixed (int* k1 = _keys)
-                {
-                    for (var i = 0; equal && i < NumInts; i++)
-                    {
-                        equal &= *(k1 + i) == *(k2 + i);
-                    }
-                }
+                throw new ArgumentOutOfRangeException(nameof(key), "Invalid key");
             }
 
-            return equal;
+            var offset = (int)key;
+            ValidateOffset(offset);
+
+            var intOffset = offset / IntSize;
+            var bitOffset = offset % IntSize;
```

Comment:
btw. just remembered, there is a faster(on most hardware, and I don't know where it is slower) approach:
`var bitOffset = offset & (IntSize - 1)`
Of course it only works if IntSize is some 2^n(where n is positive integer).
IntSize - 1 does not cost anything as IntSize is const, therefore IntSize - 1 is const as well. And & is on most systems faster than %. That is a optimization that C++ compilers do as well(don't know why C# jitter doesn't do it yet)

Reply:
`IntSize` is 5 right now so..

Reply:
Wait my bad, of course it is. No wonder I did not recognize that optimization beforehand, because it was wrong :D